iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/plan/node/write/InsertNode.java

```diff
@@ -169,6 +176,27 @@ public void setSearchIndex(long searchIndex) {
     this.searchIndex = searchIndex;
   }
 
+  public boolean isGeneratedByRemoteConsensusLeader() {
+    switch (config.getDataRegionConsensusProtocolClass()) {
+      case ConsensusFactory.IOT_CONSENSUS:
+        // NOTE: ONLY in IoTConsensus, isSyncFromLeaderWhenUsingIoTConsensus == true means this node
+        // is a follower
+        return searchIndex == ConsensusReqReader.DEFAULT_SEARCH_INDEX;
```

Comment:
Can IoTConsensus return isGeneratedByRemoteConsensusLeader as well?

Reply:
IoTConsensus has not maintain `isGeneratedByRemoteConsensusLeader` yet. For IoT, current implemention is enough